### ISSUE TITLE
HierarchicalKey throws an exception

### ISSUE THREAD

**Issue by herve26** [Bit-Wasp/bitcoin-php#778]
Hello 

I am trying to generate public keys from a master public key generated in Mycelium Wallet.
So far I have been able to do it using PHP from the command line but when I try to run the same code on my development server, it throws an exception depending on the type of key I use: ypub or xpub. 
I was able to isolate the exception and get the value that was causing the exception and apparently it because the value of the variable `$parentFingerprint` in `HierarchicalKey` becomes negative at some point.
But I don't know why it doing that and I don't know how to solve the issue.

This is the part that is causing the issue
`$serializer = new Base58ExtendedKeySerializer(new ExtendedKeySerializer($adapter, $config));`
`$key = $serializer->parse($this->network, $pub);`

**Comment by afk11** [comment]
Hey, thank's for opening this issue.

are you using 32-bit somehow (using FTP to upload files or something) Check what your local system & the server report for PHP_INT_MAX, which will tell you if you're using 32-bit. The issue re negative numbers like that happened before in a really nasty place, and it spooked us enough to disable support for 32-bit installations. 

If you're not using 32-bit, please advise, as this is unexpected :)

If you are, this isn't supported usage, and should reconsider your server or open an issue regarding lack of 32-bit support. (there hasn't been much request for this 2bh, so hard to justify the effort)

**Comment by herve26** [comment]
Thanks for getting back to me.

I have been looking at the code and I was able to narrow down the problem to the PHP function `unpack`. As stated in the documentation on the [function](http://php.net/manual/en/function.unpack.php) `unpack`:

> Caution
Note that PHP internally stores integral values as signed. If you unpack a large unsigned long and it is of the same size as PHP internally stored values the result will be a negative number even though unsigned unpacking was specified.

I have checked my installation and `PHP_INT_MAX` shows that I am using the 32-version. I will change the version that I am using to a 64-bit version in the next couple of days. 

Thanks again for the reply
